roles/ruby_s/tasks/main.yml

```diff
@@ -0,0 +1,88 @@
+---
+- name: ruby | check version
+  ansible.builtin.command: ruby -v
+  register: installed_ruby
+  ignore_errors: true
+  changed_when: false
+
+- name: ruby | install ruby if it-s not already there
+  block:
+  - name: ruby | check for ruby role apps
+    ansible.builtin.shell: dpkg-query -W 'ruby-switch'
+    ignore_errors: true
+    changed_when: false
+    register: switch
+
+  - name: ruby | uninstall global bundler
+    community.general.gem:
+      name: bundler
+      state: absent
+    when: switch is success
+
+  - name: ruby | remove installed ruby
+    ansible.builtin.apt:
+      name: ["{{ ubuntu_ruby_version }}", "{{ ubuntu_ruby_version }}-dev", "ruby-switch"]
+      state: absent
+      update_cache: true
+    when: switch is success
+
+  - name: rvm | remove the Brightbox repository
+    ansible.builtin.apt_repository:
+      repo: "ppa:brightbox/ruby-ng"
+      state: absent
+    when: switch is success
```

Comment:
This whole block seems to be about removing pre-existing artifacts. We could consider putting it in a separate file, something like `tasks/cleanup.yml` and then including it when ruby is already installed.